File: Pragma_Examples/OpenMP/Fortran/device_routine/device_routine_with_interface_v2/compute.f90
! This example was created by Johanna Potyka
! Copyright (c) 2024 AMD HPC Application Performance Team
! MIT License

      !--- device routine
      subroutine compute(x)
          !use omp_lib
!$omp declare target device_type(nohost) link(compute)
          !--------------------
          !example routine called from kernel
          !--- variables
          integer,parameter :: rk=8
          real(kind=rk),intent(inout) :: x
          !x               a value (from array)

          !--- 
          x = 1.0_rk
          

         ! write(*,*) "device_num",  omp_get_device_num()
         ! write(*,*) "is initial device",  omp_is_initial_device()
      end subroutine compute

File: Pragma_Examples/OpenMP/C/6_device_routines/3_device_routine_wdynglobaldata/1_device_routine_wdynglobaldata/compute.c
// Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
// This software is distributed under the MIT License
//
// This example was created by Johanna Potyka


#pragma omp declare target
extern double *constants;
#pragma omp end declare target

#pragma omp declare target
void compute(int cindex, double *x){
   *x = 1.0 + constants[cindex];
}
#pragma omp end declare target

